src/pool/srv_target.c

```diff
@@ -1028,6 +1028,58 @@ ds_pool_tgt_query_handler(crt_rpc_t *rpc)
 	crt_reply_send(rpc);
 }
 
+void
+ds_pool_tgt_fetch_hdls_handler(crt_rpc_t *rpc)
+{
+	struct pool_tgt_fetch_hdls_in	*in = crt_req_get(rpc);
+	struct pool_tgt_fetch_hdls_out	*out = crt_reply_get(rpc);
+	struct ds_pool			*pool;
+	int				 rc = DER_SUCCESS;
+	struct pool_iv_conn		*conn;
+
+	D_DEBUG(DF_DSMS, DF_UUID": processing rpc",
+		DP_UUID(in->tfi_pool_uuid));
+
+	pool = ds_pool_lookup(in->tfi_pool_uuid);
+	if (pool == NULL) {
+		D_ERROR("Failed to find pool "DF_UUID"\n",
+			DP_UUID(in->tfi_pool_uuid));
+		D_GOTO(out, rc = -DER_NONEXIST);
+	}
+
+	conn = in->tfi_hdls.iov_buf;
+
+	/* This while loop check first checks that there is room in the buffer
+	 * for the struct itself (not including the variable length creds field)
+	 * If there is room, it reaches into the buffer to get the size of the
+	 * creds field and repeats the check to ensure there is room
+	 */
+	while (((char *)conn - (char *)in->tfi_hdls.iov_buf +
+		sizeof(struct pool_iv_conn) <= in->tfi_hdls.iov_buf_len) &&
+	       ((char *)conn - (char *)in->tfi_hdls.iov_buf +
+		sizeof(struct pool_iv_conn) +
+		conn->pic_cred_size <= in->tfi_hdls.iov_buf_len)) {
```

Comment:
I wonder if you can use Array type RPC format, instead of analyze it yourself. Anyway, I assume this is temporary solution. probably ok for the moment.

Reply:
I wasn't aware there was an array type RPC format. I don't think we should switch though unless we can also use that same array type format for the IV code path - because that would change the inner workings of the handle flattening code, which would have to be changed back when we move to IV.